tests/system/common.py

```diff
@@ -88,6 +88,19 @@ def wait_for_mesos_endpoint(timeout_sec=timedelta(minutes=5).total_seconds()):
     return shakedown.time_wait(lambda: shakedown.mesos_available_predicate(), timeout_seconds=timeout_sec)
 
 
+def wait_for_metronome(timeout_sec=timedelta(minutes=5).total_seconds()):
+    """ Waits for the Metronome API url to be available for a given timeout.
+        It needs to succeed as many times as we have masters because that means that all adminrouters on all masters
+        are probably aware of the restarted leader.
+        Without this in multimaster setup this can be very flaky.
+    """
+
+    master_count = len(shakedown.get_all_masters())
+    shakedown.time_wait(lambda: metronome_available_predicate(),
+                        timeout_seconds=timeout_sec,
+                        required_consecutive_success_count=master_count)
```

Comment:
Hm, why do we retry as many times as we have masters?

Reply:
this is important given a frontend ELB (which is the case with AWS), which will cycle through the masters...    There are cases where you need to make sure all the masters are recovered to be in an expected working condition...   the biggest issue you are over coming by making sure that all the masters are responding ok... is to avoid flaky tests... you don't have control over which master the next call will be routed to... so if 2 out of 3 are good... call 1 works, call 2 works and call 3 fails but only because the master isn't fully up.   3 "consecutive" successes in a round robin LB model will make sure that all 3 masters are up.

Reply:
Why not check each master then? It seems like some black magic to rely on ELB cycling through all masters.